internal/controllers/addon/olm_reconciler.go

```diff
@@ -93,3 +94,54 @@ func (r *olmReconciler) Reconcile(ctx context.Context,
 func (r *olmReconciler) Name() string {
 	return OLM_RECONCILER_NAME
 }
+
+// Gets a subscription by ObjectKey
+func (r *olmReconciler) GetSubscription(
+	ctx context.Context,
+	name string,
+	namespace string,
+	destSub *operatorsv1alpha1.Subscription,
+) error {
+	if destSub == nil {
+		return errors.New("destination subscription is nil")
+	}
```

Comment:
I prefer this. I just got the impression from the code that we prefer passing pointers (I don't prefer) and also I'm not sure about returning pointers of objects created in the stack/func in go, I just googled, go creates these objects in the heap so, I will update.

Reply:
Correct, the terminology in Go would be that the  object "escapes" to the heap when you return a pointer to memory you've allocated within a func since the object cannot be garbage collected as it has a lifetime beyond the scope of the func. You generally would want to avoid this and instead let the caller control allocation so they can reuse objects/buffers for optimization.

That said for little helper methods like these it's not a real concern. Nothing should escape the scope of the reconcile loop that didn't have a longer lifetime already so we aren't introducing leaks.